proposals/extensions.md

```diff
@@ -1219,34 +1219,68 @@ implicit extension E for int
 
 # Implementation details
 
-TODO3(static) revise this to use a regular struct
+Extensions are emitted as structs with an extension marker method and an instance field for non-static extensions.  
+The type is marked with Obsolete and CompilerFeatureRequired attributes.  TODO we'll need to relax that
 
-struct Extension
-{
-  UnderlyingType underlyingValue;
-}
-// Avoid copying via Unsafe.As 
+## Marker method
 
-Extensions are implemented as ref structs with an extension marker method.  
-The type is marked with Obsolete and CompilerFeatureRequired attributes.  
 The extension marker method encodes the underlying type and base extensions as parameters in that order.  
-The marker method is called `<ImplicitExtension>$` for implicit extensions and 
+It is private and static, and is called `<ImplicitExtension>$` for implicit extensions and 
 `<ExplicitExtension>$` for explicit extensions.  
+
 For example: `implicit extension R for UnderlyingType : BaseExtension1, BaseExtension2` yields
 `private static void <ImplicitExtension>$(UnderlyingType, BaseExtension1, BaseExtension2)`.  
 
-If the extension has any instance member, then we emit a ref field (of underlying type)
-into the ref struct and a constructor.  
-TODO2(instance) The wrapping can be done with a static unspeakable factory method  
+## Instance field
 
-Values of extension types are left as values of the underlying value type, until an extension
-member is accessed. When an extension member is accessed, an extension instance is created
-with a reference to the underlying value and the member is accessed on that instance.
+If the extension not static, then we emit a private instance field of the underlying type
+into the struct.  
+We use the default layout for structs (sequential layout, with pack and size 0)
```

Comment:
Tagging @davidwrighton @jkotas to confirm that we can rely on this layout (first field is guaranteed to be at offset zero).